lambdas/bearer-token-handler/src/bearer-token-handler.ts

```diff
@@ -0,0 +1,44 @@
+import { LambdaInterface } from "@aws-lambda-powertools/commons";
+import { Logger } from "@aws-lambda-powertools/logger";
+
+const logger = new Logger();
+
+export class BearerTokenHandler implements LambdaInterface {
+  public async handler(_event: OAuthEvent, _context: unknown): Promise<any> {
+    try {
+      const url = _event.oAuthURL.value;
+      const data = {
+        client_secret: _event.totp + _event.clientSecret.value,
+        client_id: _event.clientId.value,
+        grant_type: "client_credentials",
+      } as Record<string, string>;
+
+      const formBody = Object.keys(data)
+        .map(
+          (key) => encodeURIComponent(key) + "=" + encodeURIComponent(data[key])
+        )
+        .join("&");
+
+      const response = await fetch(url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/x-www-form-urlencoded;charset=UTF-8",
+        },
+        body: formBody,
+      });
+
+      const body = (await response.json()) as OAuthResponse;
+
+      return {
+        token: body.access_token,
+        tokenExpiry: (Date.now() + body.expires_in * 1000).toString(),
+      };
+    } catch (error: any) {
```

Comment:
Maybe `unknown` here eslint would complain about `any`

Reply:
Unfortunately using `unknown` causes `error.message` to error. Might have to add `if (error.message) { do stuff... }` inside the catch block